use private templates

DasProffi opened the pull request:
## Which issues does this pull request close?
closes #859

## [OPTIONAL] Give testing instructions to reviewers

DasProffi commented:
@MaxSchaefer Do you know why the package download fails in the actions?

MaxSchaefer commented:
> @MaxSchaefer Do you know why the package download fails in the actions?

Weird. The other PRs are working fine.

DasProffi commented:
> > @MaxSchaefer Do you know why the package download fails in the actions?
> 
> Weird. The other PRs are working fine.

In this one, I updated the proto version in the package.json and that also changed the lock file

MaxSchaefer commented:
> > Weird. The other PRs are working fine.
> 
> In this one, I updated the proto version in the package.json and that also changed the lock file

@use-to wants to take a look.

use-to commented:
@MaxSchaefer Looks like we got other problems here now...

JannikWibker commented on tasks/components/cards/TaskCard.tsx:
```diff
@@ -62,7 +62,7 @@ export const TaskCard = ({
         </div>
         <div className={tw('flex flex-col gap-y-1 w-[24px]')}>
-          {assignee && <Avatar avatarUrl={assignee.avatarUrl} alt={translation.assigned} size="tiny"/>}
+          {assignee && <Avatar avatarUrl={assignee.avatarUrl ?? ''} alt={translation.assigned} size="tiny"/>}
```
Using empty strings everywhere to circumvent proper type checking doesn't seem like a good pattern

```suggestion
          {assignee && assignee.avatarUrl && <Avatar avatarUrl={assignee.avatarUrl} alt={translation.assigned} size="tiny"/>}
```

JannikWibker commented on tasks/mutations/task_template_mutations.ts:
```diff
@@ -58,14 +58,19 @@ export const useWardTaskTemplateQuery = (wardId? : string, onSuccess: (data: Tas
 }
 
-export const usePersonalTaskTemplateQuery = (createdBy? : string, onSuccess: (data: TaskTemplateDTO[]) => void = noop) => {
+export const useAllTaskTemplatesByCreator = (
+  createdBy?: string,
+  onSuccess: (data: TaskTemplateDTO[]) => void = noop,
+  onlyPrivate: boolean = false
+) => {
```
Consider changing it to `useAllTaskTemplatesByCreator({ createdBy: ..., onlyPrivate: ..., onSuccess: ... })`. Having a boolean property for which you only know the name (and thus impact) when hovering over the parameter in your editor seems like a bad idea in this case

MaxSchaefer commented:
@DasProffi pending requested changes from Jannik

MaxSchaefer reviewed (changes requested):
If I create or remote subtasks of a task-template, the task-template list does not update the subtask count, same goes for entire creation or deletion of task-templates

MaxSchaefer commented on tasks/mutations/task_template_mutations.ts:
```diff
@@ -196,7 +211,7 @@ export const useCreateMutation = (wardId: string, queryKey: QueryKey, setTemplat
 }
 
-export const useDeleteMutation = (queryKey: QueryKey, setTemplate: (task:TaskTemplateDTO | undefined) => void) => {
+export const useDeleteMutation = (queryKey: QueryKey, setTemplate: (task: TaskTemplateDTO | undefined) => void) => {
```
```suggestion
export const useDeleteMutation = (queryKey: QueryKey, setTemplate: (task?: TaskTemplateDTO) => void) => {
```

MaxSchaefer commented on tasks/mutations/task_template_mutations.ts:
```diff
@@ -88,7 +99,11 @@ export const usePersonalTaskTemplateQuery = (createdBy? : string, onSuccess: (da
+  return useAllTaskTemplatesByCreator({ createdBy, onSuccess, type: 'personalTaskTemplates' })
+}
+
+export const useUpdateMutation = (queryKey: QueryKey, setTemplate: (taskTemplate: TaskTemplateDTO | undefined) => void) => {
```
```suggestion
export const useUpdateMutation = (queryKey: QueryKey, setTemplate: (taskTemplate?: TaskTemplateDTO) => void) => {
```

MaxSchaefer commented on tasks/mutations/task_template_mutations.ts:
```diff
@@ -28,11 +28,11 @@ export type TaskTemplateDTO = {
 type QueryKey = 'personalTaskTemplates' | 'wardTaskTemplates'
 
-export const useWardTaskTemplateQuery = (wardId? : string, onSuccess: (data: TaskTemplateDTO[]) => void = noop) => {
+export const useWardTaskTemplateQuery = (wardId?: string, onSuccess: (data: TaskTemplateDTO[]) => void = noop) => {
```
Implement `useAllTaskTemplatesByCreator` as in `usePersonalTaskTemplateQuery`?
